overlays/hackage-default-args.nix

```diff
@@ -0,0 +1,58 @@
+# This overlay adds hackageDefaultArgs to provide suitable default
```

Comment:
I do kind of like the "quirks" terminology, since it suggests that this is a collection of deviations from the "normal" working case (which it is). "Hackage default args" could be anything!

Reply:
I don't suppose we can set this up so we get the quirks for anything that uses these packages? e.g. I think anything that depends on `cabal` will need a reinstallable lib GHC, right?

Reply:
I have renamed this to `quirks`.

I also looked into detecting `Cabal` and I think it will be a little tricky.  I think it might be best to leave that for another PR.

The missing `alex` tool dependencies might be fixed when we update `nix-tools` to Cabal 3.2 (I made a start on that [here](https://github.com/input-output-hk/nix-tools/tree/hkm/cabal-32)).